Source/santametricservice/SNTMetricServiceTest.m

```diff
@@ -0,0 +1,222 @@
+        @"fields" : @{
+          @"rule_type" : @[
+            @{
+              @"value" : @"binary",
+              @"created" : fixedDate,
+              @"last_updated" : fixedDate,
+              @"data" : @1,
+            },
+            @{
+              @"value" : @"certificate",
+              @"created" : fixedDate,
+              @"last_updated" : fixedDate,
+              @"data" : @2,
+            },
+          ],
+        },
+      },
+      @"/santa/rules" : @{
+        @"type" : [NSNumber numberWithInt:(int)SNTMetricTypeGaugeInt64],
+        @"fields" : @{
+          @"rule_type" : @[
+            @{
+              @"value" : @"binary",
+              @"created" : fixedDate,
+              @"last_updated" : fixedDate,
+              @"data" : @1
+            },
+            @{
+              @"value" : @"certificate",
+              @"created" : fixedDate,
+              @"last_updated" : fixedDate,
+              @"data" : @3
+            }
+          ]
+        },
+      },
+      @"/santa/using_endpoint_security_framework" : @{
+        @"type" : [NSNumber numberWithInt:(int)SNTMetricTypeConstantBool],
+        @"fields" : @{
+          @"" : @[ @{
+            @"value" : @"",
+            @"created" : fixedDate,
+            @"last_updated" : fixedDate,
+            @"data" : [NSNumber numberWithBool:YES]
+          } ]
+        }
+      },
+      @"/proc/birth_timestamp" : @{
+        @"type" : [NSNumber numberWithInt:(int)SNTMetricTypeConstantInt64],
+        @"fields" : @{
+          @"" : @[ @{
+            @"value" : @"",
+            @"created" : fixedDate,
+            @"last_updated" : fixedDate,
+            @"data" : [NSNumber numberWithLong:1250999830800]
+          } ]
+        },
+      },
+      @"/proc/memory/virtual_size" : @{
+        @"type" : [NSNumber numberWithInt:(int)SNTMetricTypeGaugeInt64],
+        @"fields" : @{
+          @"" : @[ @{
+            @"value" : @"",
+            @"created" : fixedDate,
+            @"last_updated" : fixedDate,
+            @"data" : [NSNumber numberWithInt:987654321]
+          } ]
+        }
+      },
+      @"/proc/memory/resident_size" : @{
+        @"type" : [NSNumber numberWithInt:(int)SNTMetricTypeGaugeInt64],
+        @"fields" : @{
+          @"" : @[ @{
+            @"value" : @"",
+            @"created" : fixedDate,
+            @"last_updated" : fixedDate,
+            @"data" : [NSNumber numberWithInt:123456789]
+          } ]
+        },
+      },
+    }
+  };
+}
+
+- (void)setUp {
+  [self initializeValidMetricsDict];
+  // create the configurator
+  self.mockConfigurator = OCMClassMock([SNTConfigurator class]);
+  OCMStub([self.mockConfigurator configurator]).andReturn(self.mockConfigurator);
+
+  // create a temp dir
+  char template[] = "/tmp/sntmetricsservicetestdata.XXXXXXX";
+  char *tempPath = mkdtemp(template);
+
+  if (tempPath == NULL) {
+    NSLog(@"Unable to make temp directory");
+    exit(1);
+  }
```

Comment:
```suggestion
  XCTAssertNotEqual(tempPath, NULL, @"Unable to make temp dir");
```

setUp and tearDown are considered part of the test so you can use XCTAssert* macros in them to get the output formatted nicely in Xcode (& xcodebuild)